systemtest/src/test/java/io/strimzi/systemtest/operators/MultipleClusterOperatorsST.java

```diff
@@ -0,0 +1,247 @@
+                .addToConfig("topics", TOPIC_NAME)
+            .endSpec()
+            .build());
+
+        KafkaBasicExampleClients basicClients = new KafkaBasicExampleClients.Builder()
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+            .withTopicName(TOPIC_NAME)
+            .withMessageCount(MESSAGE_COUNT)
+            .build();
+
+        basicClients.createAndWaitForReadiness(basicClients.producerStrimzi().build());
+        ClientUtils.waitForClientSuccess(producerName, DEFAULT_NAMESPACE, MESSAGE_COUNT);
+
+        KafkaConnectUtils.waitForMessagesInKafkaConnectFileSink(kafkaConnectPodName, Constants.DEFAULT_SINK_FILE_PATH, "Hello-world - 99");
+    }
+
+    @Test
+    void testLabelingAndUnlabelingOfResource() {
+        int scaleTo = 4;
+        deployCOInNamespace(FIRST_CO_NAME, DEFAULT_NAMESPACE, FIRST_CO_SELECTOR_ENV, false);
+        deployCOInNamespace(SECOND_CO_NAME, DEFAULT_NAMESPACE, SECOND_CO_SELECTOR_ENV, false);
+
+        LOGGER.info("Deploying Kafka with CR selector of first CO");
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(clusterName, 3)
+            .editOrNewMetadata()
+                .addToLabels(FIRST_CO_SELECTOR)
+            .endMetadata()
+            .build());
+
+        LOGGER.info("Removing CR selector from Kafka and increasing number of replicas to 4, new pod should not appear");
+        KafkaResource.replaceKafkaResource(clusterName, kafka -> {
+            kafka.getMetadata().getLabels().clear();
+            kafka.getSpec().getKafka().setReplicas(scaleTo);
+        });
+        KafkaUtils.waitForClusterStability(clusterName);
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName));
+        Map<String, String> zkPods = StatefulSetUtils.ssSnapshot(KafkaResources.zookeeperStatefulSetName(clusterName));
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(KafkaResources.entityOperatorDeploymentName(clusterName));
+
+        LOGGER.info("Adding CR selector of second CO to Kafka");
+        KafkaResource.replaceKafkaResource(clusterName, kafka -> kafka.getMetadata().setLabels(SECOND_CO_SELECTOR));
+
+        LOGGER.info("Waiting for rolling update because of new label, also Kafka should scale to {}", scaleTo);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.zookeeperStatefulSetName(clusterName), 3, zkPods);
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(clusterName), scaleTo, kafkaPods);
+        DeploymentUtils.waitTillDepHasRolled(KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
+
+        Kafka kafka = KafkaResource.kafkaClient().inNamespace(DEFAULT_NAMESPACE).withName(clusterName).get();
+        assertThat(kafka.getSpec().getKafka().getReplicas(), is(scaleTo));
```

Comment:
I'm not sure we need to test it here => you changed number in spec on line 158, nobody would change that back. But it does not mean it is scaled. 

We need to check it really scaled instead => for example by listing the pods and counting them or just checking the pod with the name and correct index exists.